sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -110,6 +126,17 @@ static event_queue_t _queue;
 static uint8_t _listen_buf[CONFIG_GCOAP_PDU_BUF_SIZE];
 static sock_udp_t _sock_udp;
 
+#if IS_ACTIVE(CONFIG_GCOAP_ENABLE_DTLS)
+/* DTLS variables and definitions */
+#define SOCK_DTLS_CLIENT_TAG (2)
+
+static sock_dtls_t _sock_dtls;
```

Comment:
What if instead we did

```patch
--- a/pkg/tinydtls/include/sock_dtls_types.h
+++ b/pkg/tinydtls/include/sock_dtls_types.h
@@ -39,8 +39,8 @@ extern "C" {
  * @brief Information about DTLS sock
  */
 struct sock_dtls {
+    sock_udp_t udp_sock;                    /**< Underlying UDP sock to use */
     dtls_context_t *dtls_ctx;               /**< TinyDTLS context for sock */
-    sock_udp_t *udp_sock;                   /**< Underlying UDP sock to use */
 #if defined(SOCK_HAS_ASYNC) || defined(DOXYGEN)
     /**
      * @brief   Network stack internal buffer context.
```

And use the `.flags` field to indicate there is a DTLS context attached to the socket? 

Could that save the application from dealing with DTLS entirely (save for calling a DTLS setup function on the DTLS socket) and leave the rest to GNRC?